### PR TITLE
Set auto-added artifacts optional in recurring run (job)

### PR DESCRIPTION
**Pull request by jingzhang36**
We set them optional already for one-time run (https://github.com/kubeflow/pipelines/pull/1289). We should also set them for recurring run.

Manual test on kfp deployment.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3041)

### CODE REVIEW

File: backend/src/apiserver/resource/resource_manager.go [hunk 1]
```diff
@@ -535,6 +535,18 @@ func (r *ResourceManager) CreateJob(apiJob *api.Job) (*model.Job, error) {
 			},
 		},
 	}
+
+	// Marking auto-added artifacts as optional. Otherwise most older workflows will start failing after upgrade to Argo 2.3.
+	// TODO: Fix the components to explicitly declare the artifacts they really output.
+	// TODO: Change the compiler to stop auto-adding those two atrifacts to all tasks.
```
Comment (numerology) [inline backend/src/apiserver/resource/resource_manager.go]: IIRC the compiler no longer adds those two artifact to all tasks. @Ark-kun can you correct me if I'm wrong?

Comment (Ark-kun) [reply backend/src/apiserver/resource/resource_manager.go]: Yes. The compiler stopped doing this in SDK v0.1.29: https://github.com/kubeflow/pipelines/pull/2046.
Pipelines compiled by older SDKs have the artifacts hardcoded in them.

Comment (IronPan) [reply backend/src/apiserver/resource/resource_manager.go]: Does this todo still apply?

Comment (numerology) [reply backend/src/apiserver/resource/resource_manager.go]: I believe no. That's why I was asking :)

Comment (jingzhang36) [reply backend/src/apiserver/resource/resource_manager.go]: Great to know. I'll remove the complier TODO, and also in the compiler TODO in the one-time run.

**Pull request merged**